dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/PeriodType.java

```diff
@@ -344,6 +344,13 @@ public Period toIsoPeriod( DateTimeUnit dateTimeUnit )
      */
     public abstract int getFrequencyOrder();
 
+    /**
+     * Returns a string usable as the interval value in PostgreSQL queries.
+     *
+     * @return the SQL interval.
+     */
+    public abstract String getSqlInterval();
```

Comment:
I'm not sure about this: we are now adding sql (postgres specific) logic to a domain class, which should be agnostic regarding the underlying persistent engine. I wonder if it would be better to have a separate utility class that would return this interval String based on the given period.

Reply:
I'd be interested in another opinion about this also. I agree with the principle. The main reason I did it this way is so that if/when we implement a new PeriodType we won't forget to put this code in. (And we are now assuming postgres in some other parts of the code.)

Reply:
We have comitted to Postgres, so I think it's fine to do some postgres specific stuff in places. @luciano-fiandesio would it make more sense to make this an enum, that would contain the postgres specific value, or a utility class that converted from the enum to the postgres value? I am open for both.

Reply:
@stian-sandvold is not really about Postgres vs. any other DB, but is more adding DB specific logic to domain classes (which are growing in complexity since we are adding new functionalities). In this case, I agree it's not really logic, but more like a constant, but still, from a single-responsibility concept point of view, I think a simple utility class would be better (e.g. `SqlIntervalUtil.getInterval(Period p)`)